x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.test.ts

```diff
@@ -330,7 +332,70 @@ describe('Host Isolation', () => {
   });
 
   describe('Cases', () => {
-    it.todo('logs a comment to the provided case');
-    it.todo('logs a comment to any cases associated with the given alert');
+    let casesClient: CasesClientMock;
+
+    beforeEach(async () => {
+      casesClient = (await endpointAppContextService.getCasesClient(
+        {} as KibanaRequest
+      )) as CasesClientMock;
+
+      let counter = 1;
+      casesClient.cases.getCasesByAlertID.mockImplementation(async () => {
+        return [
+          {
+            id: `case-${counter++}`,
+            title: 'case',
+          },
+        ];
+      });
+    });
+
+    it('logs a comment to the provided case', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], case_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenCalledTimes(2);
+    });
+
+    it('logs a comment to any cases associated with the given alerts', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], alert_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenNthCalledWith(
+        1,
+        expect.objectContaining({ caseId: 'case-1' })
+      );
+      expect(casesClient.attachments.add).toHaveBeenNthCalledWith(
+        2,
+        expect.objectContaining({ caseId: 'case-2' })
+      );
+    });
+
+    it('logs a comment to any cases  provided on input along with cases associated with the given alerts', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], case_ids: ['ONE', 'TWO', 'TWO'], alert_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenCalledTimes(4);
```

Comment:
Is this specifically testing `case_ids` dedupe behavior? If it is, that isn't in the test description as what's being tested. This test is then already overloaded with what it says it's testing ( `case_ids` + `alert_ids` ). So that should be a separate test.

That's also not guaranteed spec behavior, so I wouldn't love seeing a test for that anyway

Reply:
Not sure what you mean by spec behavior, but this tests an area of the code that is not currently accounted for. both case_ids and alert_ids can be used and those could have duplicate case ids (ex. I could have made one of the case_ids be `case-1`.  I can add the word "dedupe" to the description if that helps, but I personally did not find it to be needed

Reply:
You are testing this by passing two identical `case_ids` (`"TWO"`). It is not guaranteed to dedupe _that_. Rather, if you want to test that a matching `case_id` and the *same* case_id from a given `alert_id`, then one of your `case_ids` should be `"case-1"`, right? So that you are explicitly passing a `case_id` that would also get returned in a lookup call from the `alert_ids`.